content-scripts/cs.js

```diff
@@ -377,6 +377,19 @@ async function onInfoAvailable({ globalState: globalStateMsg, addonsWithUserscri
 
 const escapeHTML = (str) => str.replace(/([<>'"&])/g, (_, l) => `&#${l.charCodeAt(0)};`);
 
+if (location.pathname.match(/\/discuss\/(.*)/gm)) { // First as scratchblocks2 runs fast, we need to preserve original blocks.
+  const preserveBlocks = () => {
+    document.querySelectorAll("pre.blocks").forEach((el) => {
+      el.setAttribute("data-original", el.innerText);
+    });
+  };
+  if (document.readyState !== "loading") {
+    (async () => preserveBlocks())(); // Run this asynchronously to run below scripts faster
```

Comment:
That's not how JavaScript works.

Reply:
I'll tweak the code later, please keep the code unchanged.
But keep in mind, that is still called synchronously.

Reply:
I thought that's how async works tho but ok

Reply:
You probably wanted to do setTimeout of 0 or queueMicrotask. Async functions are still called synchronously in JavaScript. Only things that are truly async are setTimeout, queueMicrotask, and browser APIs like fetch.

Reply:
i'll change it to settimeout i guess